toolkit/tools/pkg/imagecustomizerlib/customizeboot.go

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package imagecustomizerlib
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/safechroot"
+)
+
+var (
+	linuxCommandLineRegex = regexp.MustCompile(`\tlinux .* (\$kernelopts)`)
+)
+
+func handleKernelCommandLine(commandLineAdd string, imageChroot *safechroot.Chroot) error {
+	var err error
+
+	if commandLineAdd == "" {
+		// Nothing to do.
+		return nil
+	}
+
+	grub2ConfigFilePath := filepath.Join(imageChroot.RootDir(), "/boot/grub2/grub.cfg")
+
+	// Read the existing grub.cfg file.
+	grub2ConfigFileBytes, err := os.ReadFile(grub2ConfigFilePath)
+	if err != nil {
+		return fmt.Errorf("failed to read existing grub2 config file: %w", err)
+	}
+
+	grub2ConfigFile := string(grub2ConfigFileBytes)
+
+	// Find the point where the new command line arguments should be added.
+	match := linuxCommandLineRegex.FindStringSubmatchIndex(grub2ConfigFile)
```

Comment:
More brainstorming, is there value in moving to a dedicated variable for `root=` etc. so we don't stomp the disks? Then we can just replace/add after that well-known variable.

Not useful for M2, but maybe M3 feature?

Reply:
At least in the grub2-mkconfig case, root is supplied separately. 

Example in kernel-mshv, which uses the same logic as the standard 10_linux template: https://github.com/microsoft/CBL-Mariner/pull/6343/files#diff-6d30214896e767042064c1c3dafc8beb64ce2e096b1f5245fce15a57a687f0e0R64